src/bit_set.rs

```diff
@@ -85,20 +82,47 @@ impl Bitset {
         } else {
             let word = bit / bits_per_word() - 1;
             let tail = self.tail();
-            if word >= tail.len() {
+            word < tail.len() && (tail[word] & (1 << (bit & (bits_per_word() - 1)))) != 0
+        }
+    }
+
+    /// Adds a single bit to a set, and returns the old value.  Equivalent to
+    /// `{ let old = bitset.has_bit(bit); bitset.set_bit(bit); old }`.
+    pub fn set_bit_if(&mut self, bit: usize) -> bool {
+        if bit < bits_per_word() {
+            let old = (self.head & (1 << bit)) != 0;
+            self.head |= 1 << bit;
+            old
+        } else {
+            let word = bit / bits_per_word() - 1;
+            let tail = self.tail_mut();
+            let mask = 1 << (bit & (bits_per_word() - 1));
+            let old = if word >= tail.len() {
+                tail.resize(word + 1, 0);
                 false
             } else {
-                (tail[word] & (1 << (bit & (bits_per_word() - 1)))) != 0
-            }
+                (tail[word] & mask) != 0
+            };
+            tail[word] |= mask;
+            old
+        }
+    }
+
+    /// same as `self <= rhs`, but with `rhs` treated as zero if empty
+    #[inline]
+    pub fn le_opt(&self, rhs: Option<&Bitset>) -> bool {
```

Comment:
Reject unless we have a compelling argument better than "someone might need this someday".  Every line of code is a maintenance burden